flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/physical/batch/RemoveRedundantLocalHashAggRule.java

```diff
@@ -0,0 +1,104 @@
+                        // removed,
+                        // see BatchPhysicalHashAggRule for details.
+                        localAgg.getAggCallToAggFunction(),
+                        false);
+        call.transformTo(newGlobalAgg);
+    }
+
+    /** Configuration for {@link RemoveRedundantLocalHashAggRule}. */
+    @Value.Immutable(singleton = false)
+    public interface RemoveRedundantLocalHashAggRuleConfig extends RelRule.Config {
+        RemoveRedundantLocalHashAggRule.RemoveRedundantLocalHashAggRuleConfig DEFAULT =
+                ImmutableRemoveRedundantLocalHashAggRule.RemoveRedundantLocalHashAggRuleConfig
+                        .builder()
+                        .build()
+                        .withOperandSupplier(
+                                b0 ->
+                                        b0.operand(BatchPhysicalHashAggregate.class)
+                                                .oneInput(
+                                                        b1 ->
+                                                                b1.operand(
+                                                                                BatchPhysicalLocalHashAggregate
+                                                                                        .class)
+                                                                        .oneInput(
+                                                                                b2 ->
+                                                                                        b2.operand(
+                                                                                                        RelNode
+                                                                                                                .class)
+                                                                                                .oneInput(
+                                                                                                        b3 ->
+                                                                                                                b3.operand(
+                                                                                                                                FlinkConventions
+                                                                                                                                        .BATCH_PHYSICAL()
+                                                                                                                                        .getInterface())
+                                                                                                                        .noInputs()))))
+                        .withDescription("RemoveRedundantLocalHashAggRule")
+                        .as(
+                                RemoveRedundantLocalHashAggRule
+                                        .RemoveRedundantLocalHashAggRuleConfig.class);
```

Comment:
@snuyanzin why need this 
                    .as(
                                RemoveRedundantLocalHashAggRule
                                        .RemoveRedundantLocalHashAggRuleConfig.class);?